oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/storage/PersistenceTimer.java

```diff
@@ -63,98 +70,150 @@ public void start(ModuleManager moduleManager, CoreModuleConfig moduleConfig) {
-            MetricsTag.EMPTY_KEY, MetricsTag.EMPTY_VALUE
+                "persistence_timer_bulk_prepare_latency", "Latency of the prepare stage in persistence timer",
+                MetricsTag.EMPTY_KEY, MetricsTag.EMPTY_VALUE
         );
         executeLatency = metricsCreator.createHistogramMetric(
-            "persistence_timer_bulk_execute_latency", "Latency of the execute stage in persistence timer",
-            MetricsTag.EMPTY_KEY, MetricsTag.EMPTY_VALUE
+                "persistence_timer_bulk_execute_latency", "Latency of the execute stage in persistence timer",
+                MetricsTag.EMPTY_KEY, MetricsTag.EMPTY_VALUE
+        );
+        allLatency = metricsCreator.createHistogramMetric(
+                "persistence_timer_bulk_all_latency", "Latency of the all stage in persistence timer",
+                MetricsTag.EMPTY_KEY, MetricsTag.EMPTY_VALUE
         );
+
         syncOperationThreadsNum = moduleConfig.getSyncThreads();
         maxSyncoperationNum = moduleConfig.getMaxSyncOperationNum();
+        batchExecutorService = Executors.newSingleThreadExecutor();
         executorService = Executors.newFixedThreadPool(syncOperationThreadsNum);
+        prepareExecutorService = Executors.newFixedThreadPool(moduleConfig.getPrepareThreads());
         if (!isStarted) {
             Executors.newSingleThreadScheduledExecutor()
-                     .scheduleWithFixedDelay(
-                         new RunnableWithExceptionProtection(() -> extractDataAndSave(batchDAO), t -> log
-                             .error("Extract data and save failure.", t)), 5, moduleConfig.getPersistentPeriod(),
-                         TimeUnit.SECONDS
-                     );
+                    .scheduleWithFixedDelay(
+                            new RunnableWithExceptionProtection(() -> extractDataAndSave(batchDAO), t -> log
+                                    .error("Extract data and save failure.", t)), 5, moduleConfig.getPersistentPeriod(),
+                            TimeUnit.SECONDS
+                    );
 
             this.isStarted = true;
         }
     }
 
-    private void extractDataAndSave(IBatchDAO batchDAO) {
+    @VisibleForTesting
+    void extractDataAndSave(IBatchDAO batchDAO) {
         if (log.isDebugEnabled()) {
             log.debug("Extract data and save");
         }
 
         long startTime = System.currentTimeMillis();
+        HistogramMetrics.Timer allTimer = allLatency.createTimer();
 
         try {
-            HistogramMetrics.Timer timer = prepareLatency.createTimer();
+            List<PersistenceWorker> persistenceWorkers = new ArrayList<>();
+            persistenceWorkers.addAll(TopNStreamProcessor.getInstance().getPersistentWorkers());
+            persistenceWorkers.addAll(MetricsStreamProcessor.getInstance().getPersistentWorkers());
+            CountDownLatch countDownLatch = new CountDownLatch(MetricsStreamProcessor.getInstance().getPersistentWorkers().size());
```

Comment:
I believe it should be

```suggestion
            CountDownLatch countDownLatch = new CountDownLatch(persistenceWorkers.size());
```